lib/src/external_module.cc

```diff
@@ -342,8 +346,10 @@ ActionResponse ExternalModule::callNonBlockingAction(const ActionRequest& reques
     LOG_TRACE("Input for the %1%: %2%", request.prettyLabel(), input_txt);
 
     auto exec = lth_exec::execute(
-#ifdef _WIN32
+#if defined(_WIN32)
         "cmd.exe", { "/c", path_, action_name },
+#elif defined(__sun)
+        "/usr/bin/ctrun", { "-l", "child", path_, action_name },
```

Comment:
I'm not sure whether passing the absolute path of ctrun is the right choice.
facter does that for Solaris' zonename: https://github.com/puppetlabs/facter/blob/master/lib/src/facts/solaris/zone_resolver.cc#L17.
Should we rely on leatherman's which instead and only pass the executable name?
@MikaelSmith , @branan  thoughts?